### PR TITLE
[AF-1430] add secure settings sample app

### PR DESCRIPTION
**Pull request by token-cjg**
There are currently no secure settings sample apps for v2.  Uh oh, looks like the cat is out of the bag! 🙀  

...let's create one and add it to this repository.

### Description

Uses the cat api.

You can sign up for an api token at thecatapi.com.  Note that this token needs to go in the `api_token` field.  (The other secure setting is irrelevant.)

### Screenshot

screen shot 2018-11-28 at 5 11 57 pm

### References

* https://zendesk.atlassian.net/browse/AF-1430

### CODE REVIEW

File: v2/support/secure_settings_sample_app/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# App name
+
+Bag End Catering Helper
```
Comment (hoangtrvu) [inline v2/support/secure_settings_sample_app/README.md]: I think this README file needs a bit more polishing to be useful 😄You can see other READMEs of current sample apps for inspiration.

File: v2/support/secure_settings_sample_app/assets/iframe.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    var client = ZAFClient.init();
+    client.invoke('resize', { width: '100%', height: '200px' });
+    var settings = {
+      url: 'https://api.thecatapi.com/v1/images/search?format=json',
+      headers: {"x-api-key": "{{setting.catering_token}}"},
+      secure: true,
+      type: 'GET',
+      contentType: 'application/json'
+    };
+    client.request(settings).then(function(data) { console.log(data); });
```
Comment (hoangtrvu) [inline v2/support/secure_settings_sample_app/assets/iframe.html]: These calls should be wrapped inside `app.registered` callback. For example:
```suggestion
    var client = ZAFClient.init();

    client.on('app.registered', function() {
      var settings = { ... };

      client.invoke('resize', { width: '100%', height: '200px' });
      client.request(settings).then(function(data) {
        // console.log(data);
      });
    });
```

File: v2/support/secure_settings_sample_app/manifest.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "Bag End Catering Helper",
+  "author": {
+    "name": "Frodo Baggins",
+    "email": "cgoddard@zendesk.com",
+    "url": "https://hobbiton.zendesk-staging.com"
```
Comment (hoangtrvu) [inline v2/support/secure_settings_sample_app/manifest.json]: You probably don't want to give out your email and this URL here. Maybe can use a generic one like [this](https://github.com/zendesk/demo_apps/blob/master/v2/support/basic_ticket_sample_app/manifest.json) instead.

File: v2/support/secure_settings_sample_app/manifest.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    }
+  },
+  "parameters": [
+    {
+      "name": "catering_token",
+      "type": "text",
+      "secure": true
+    },
+    {
+      "name": "the_one_broom_id",
```
Comment (hoangtrvu) [inline v2/support/secure_settings_sample_app/manifest.json]: this is not used, can be removed

File: v2/support/secure_settings_sample_app/manifest.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    "url": "https://hobbiton.zendesk-staging.com"
+  },
+  "defaultLocale": "en",
+  "private": true,
+  "location": {
+    "support": {
+      "ticket_sidebar": "assets/iframe.html"
+    }
+  },
+  "parameters": [
```
Comment (hoangtrvu) [inline v2/support/secure_settings_sample_app/manifest.json]: you need to define `domainWhitelist` in this manifest too. See [doc](https://developer.zendesk.com/apps/docs/developer-guide/using_sdk#using-secure-settings).

File: v2/support/secure_settings_sample_app/manifest.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    "url": "https://support.zendesk.com"
+  },
+  "defaultLocale": "en",
+  "private": true,
+  "location": {
+    "support": {
+      "ticket_sidebar": "assets/iframe.html"
+    }
+  },
+  "parameters": [
```
Comment (danielbreves) [inline v2/support/secure_settings_sample_app/manifest.json]: We should also include a [`domainWhitelis`](https://developer.zendesk.com/apps/docs/developer-guide/using_sdk#using-secure-settings).

Comment (token-cjg) [reply v2/support/secure_settings_sample_app/manifest.json]: Ooops.  I'll get to it.

File: v2/support/secure_settings_sample_app/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+# Bag End Catering Helper
```
Comment (danielbreves) [inline v2/support/secure_settings_sample_app/README.md]: 😅 I think the humour here may be a bit distracting and could cause some confusion. Maybe we should prefer a more plain readme for consistency with the other demo apps?

Comment (token-cjg) [reply v2/support/secure_settings_sample_app/README.md]: Nonsense, this is a perfectly serious Readme! What are you talking about ?!

I might tidy it up just a little, though.

File: v2/support/secure_settings_sample_app/README.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# Secure Settings Example App
+
+This sample app shows how to add Secure Settings to an app. "Secure Settings" are a neat feature that allows your App to pass a secure setting via the proxy that is invisible to normal agents.
```
Comment (Mehul-Porwal) [inline v2/support/secure_settings_sample_app/README.md]: `is` a neat feature

File: .gitignore [hunk 9]
```diff
@@ -1,3 +1,4 @@
 *.zip
 **/.zat
 **/tmp
+.DS_Store
```
Comment (danielbreves) [inline .gitignore]: Usually this is ignored at the user folder level.

File: v2/support/secure_settings_sample_app/README.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
+# Secure Settings Example App
+
+This sample app shows how to add Secure Settings to an app. "Secure Settings" is a neat feature that allows your App to pass a secure setting via the proxy that is invisible to normal agents.
```
Comment (danielbreves) [inline v2/support/secure_settings_sample_app/README.md]: Could we include a link to the docs here? https://developer.zendesk.com/apps/docs/developer-guide/using_sdk#using-secure-settings

File: v2/support/secure_settings_sample_app/README.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+In particular, this App uses a secure setting to store a token for https://thecatapi.com/ which is required for making api requests.  To get such a key, you will need to sign up for an account, and then grab your api key from the signup email.
+
+On installation the account admin should paste this key into the "catering_token" field.
+
+The app will log to the browser console a single response from thecatapi, which is a GET request to /images/search per https://documenter.getpostman.com/view/4016432/RWToRJCq .
```
Comment (danielbreves) [inline v2/support/secure_settings_sample_app/README.md]: Could you update this to match the new functionality?

File: .gitignore (hunk 9 above)
Comment (token-cjg) [reply .gitignore]: Right, I was wondering why this wasn't included.

File: v2/support/secure_settings_sample_app/README.md (hunk 11 above)
Comment (token-cjg) [reply v2/support/secure_settings_sample_app/README.md]: Good point, done.

File: v2/support/secure_settings_sample_app/README.md (hunk 10 above)
Comment (token-cjg) [reply v2/support/secure_settings_sample_app/README.md]: Yes, I should do that! Done.

**Pull request merged**